pkg/hnsw/hnsw_test.go

```diff
@@ -559,4 +557,45 @@ func TestHnsw_KnnSearch(t *testing.T) {
 			t.Fatalf("expected closest points to be %v, got %v", expectedIds, gotIds)
 		}
 	})
+
+	t.Run("cluster c search", func(t *testing.T) {
+		clusterC := append(append([]Point{}, clusterA...), clusterB...)
+
+		clusterCLen := len(clusterC)
+
+		h := NewHnsw(2, clusterCLen, clusterCLen, Point{0, 0})
```

Comment:
This means when we `Insert` every vector, the graph will consider every point as a candidate. In addition, every point's number of neighbors is `clusterCLen`.